src/test/project-manager.test.ts

```diff
@@ -70,4 +75,46 @@ describe('ProjectManager', () => {
 			assert.equal('/src/jsconfig.json', jsConfig.configFilePath);
 		});
 	});
+	describe('didOpen()', () => {
+		beforeEach(async () => {
+			diagnosticsSpy = sinon.spy();
+			diagnosticsPublisher.updateFileDiagnostics = diagnosticsSpy;
+			memfs = new InMemoryFileSystem('/');
+			const localfs = new MapFileSystem(new Map([
+				['file:///package.json', '{"name": "package-name-1"}'],
+				['file:///src/dummy.ts', 'const num: number = "banana";']
+			]));
+			const updater = new FileSystemUpdater(localfs, memfs);
+			projectManager = new ProjectManager('/', memfs, updater, diagnosticsPublisher, true);
+			await projectManager.ensureAllFiles();
+		});
+		it('should compile opened file and return diagnostics', async () => {
+			projectManager.didOpen('file:///src/dummy.ts', 'const num: number = "banana";');
+			const lastArgs = diagnosticsSpy.lastCall.args[0];
+			assert.equal(lastArgs.length, 1);
```

Comment:
Added an assert as suggested, but I couldn't find a way to assert on the last call only with sinon's asserts. They suggest using `sinon.assert.calledWith`, and we would need a `sinon.match` to avoid check all of TS's  diagnostics type. I tried this briefly but ran out of time, let me know if you want me to switch it to this syntax.